### PR TITLE
[XTZ] Add compressed SECP256K1 keys support

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR adds Tezos SECP256K1 compressed pub key support in addition to the existing raw SECP256K1 pub key support.

### CODE REVIEW

**Review by pollastri-pierre — APPROVED** [review]

**Review by gagbo — APPROVED** [review]
Thanks. I have a small nitpick but not really important

File: core/src/tezos/TezosLikeAddress.cpp [hunk 1]
```diff
@@ -137,17 +137,30 @@ namespace ledger {
...
+                        return BLAKE::blake2b(
+                                vector::concat(
+                                        std::vector<uint8_t>{static_cast<uint8_t>(pubKey[00])},
+                                        std::vector<uint8_t>{pubKey.begin() + 1, pubKey.begin() + 33}
+                                ),
+                                20
+                        );
+                    } else {
+                        throw Exception(api::ErrorCode::INVALID_ARGUMENT,
+                                        "Invalid SECP256k1 or P256 public key: should be 33 or 65 bytes.");
```
Comment (gagbo) [inline core/src/tezos/TezosLikeAddress.cpp:160]: if branches that return don't need else if

File: core/src/tezos/TezosLikeAddress.cpp [hunk 2]
```diff
@@ -137,17 +137,30 @@ namespace ledger {
...
+                    if (pubKey.size() == 65) {
+                        // Uncompressed key
+                        return BLAKE::blake2b(
+                                vector::concat(
+                                        std::vector<uint8_t>{static_cast<uint8_t>(0x02 + (pubKey[64] & 0x01))},
+                                        std::vector<uint8_t>{pubKey.begin() + 1, pubKey.begin() + 33}
+                                ),
+                                20
+                        );
+                    } else if (pubKey.size() == 33) {
```
Comment (gagbo) [inline core/src/tezos/TezosLikeAddress.cpp:149]:
```suggestion
                    if (pubKey.size() == 33) {
```

File: core/src/tezos/TezosLikeAddress.cpp [hunk 3]
```diff
@@ -137,17 +137,30 @@ namespace ledger {
...
+                                vector::concat(
+                                        std::vector<uint8_t>{static_cast<uint8_t>(pubKey[00])},
+                                        std::vector<uint8_t>{pubKey.begin() + 1, pubKey.begin() + 33}
+                                ),
+                                20
+                        );
+                    } else {
+                        throw Exception(api::ErrorCode::INVALID_ARGUMENT,
+                                        "Invalid SECP256k1 or P256 public key: should be 33 or 65 bytes.");
                     }
```
Comment (gagbo) [inline core/src/tezos/TezosLikeAddress.cpp:161]:
```suggestion
                    
                    throw Exception(api::ErrorCode::INVALID_ARGUMENT,
                                    "Invalid SECP256k1 or P256 public key: should be 33 or 65 bytes.");
```

**Review by alekece — APPROVED** [review]

**Comment by noirbizarre** [comment]
Changes applied :+1:

**Comment by twilgenbus-ledger** [comment]
This PR's code was merged elsewhere to 4.1.0

**Pull request closed without merging**